compiler/optimizer/LocalOpts.cpp

```diff
@@ -8166,8 +8166,15 @@ TR_ColdBlockMarker::identifyColdBlocks()
    for (TR::AllBlockIterator iter(optimizer()->getMethodSymbol()->getFlowGraph(), comp()); iter.currentBlock(); ++iter)
       {
       TR::Block *block = iter.currentBlock();
+
       if (block->isCold())
+         {
+         // OSR blocks may not have had a chance to set their frequencies yet
+         if (block->isOSRCodeBlock() || block->isOSRCatchBlock())
+            block->setFrequency(0);
```

Comment:
Since the cold block frequencies mean something special (ie the source) I think there is a number reserved for ilgen or similar - perhaps that number should be used here.